compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/FoldConstantLowering.kt

```diff
@@ -176,6 +174,18 @@ class FoldConstantLowering(private val context: JvmBackendContext) : IrElementTr
                     else -> expression
                 }
             }
+
+            override fun visitStringConcatenation(expression: IrStringConcatenation): IrExpression {
+                expression.transformChildrenVoid(this)
+                return if (expression.arguments.all { it is IrConst<*> && it.value is String }) {
```

Comment:
Would also like to see `Char` handled.

And really so long as the stdlib version is equal to or less than the compiler version all constants should be able to be supported.

Reply:
@pyos Really why not support all types,e.g.:
val p = "123${1}"?

Reply:
Yeah, good point, fixed that.